### PR TITLE
Update APIs of add-ons and core

### PR DESCRIPTION
**Pull request by thc202**
Update core APIs for 2.14.0.

Update the APIs of the following add-on:
 - Selenium version 15.15.0.

Prepare release.
Test with newer Python version.

### CODE REVIEW

**Comment by thc202** [comment]
Removed version 3.12, will require build changes.

**Review by kingthorin — APPROVED** [review]

**Review by ricekot — APPROVED** [review]
Thanks!

File: src/zapv2/core.py [hunk 1]
```diff
@@ -693,3 +699,15 @@ def send_har_request(self, request, followredirects=None, apikey=''):
...
             params['followRedirects'] = followredirects
         return (self.zap._request_other(self.zap.base_other + 'core/other/sendHarRequest/', params))
+
+    def file_download(self, filename, apikey=''):
+        """
+        Download a file from the transfer directory
+        """
+        return (self.zap._request_other(self.zap.base_other + 'core/other/fileDownload/', {'fileName': filename, 'apikey': apikey}))
+
+    def file_upload(self, filename, filecontents, apikey=''):
```
Comment (ricekot) [inline src/zapv2/core.py:709]: Worth testing once to see if this works as expected...

Comment (thc202) [reply src/zapv2/core.py:709]: It doesn't :(

Comment (thc202) [reply src/zapv2/core.py:709]: Now using the expected HTTP method.

**Comment by thc202** [comment]
Ready for another review.

File: src/zapv2/core.py (hunk 1 above)
Comment (ricekot) [reply src/zapv2/core.py:709]: Great, thank you!

**Review by kingthorin — APPROVED** [review]

**Review by ricekot — APPROVED** [review]
Thanks!

**Pull request merged**